internal/services/sentinel/sentinel_alert_rule_scheduled_resource.go

```diff
@@ -151,11 +151,11 @@ func resourceSentinelAlertRuleScheduled() *pluginsdk.Resource {
 									"entity_matching_method": {
 										Type:     pluginsdk.TypeString,
 										Optional: true,
-										Default:  securityinsight.EntitiesMatchingMethodNone,
+										Default:  securityinsight.MatchingMethodAnyAlert,
 										ValidateFunc: validation.StringInSlice([]string{
-											string(securityinsight.EntitiesMatchingMethodAll),
-											string(securityinsight.EntitiesMatchingMethodCustom),
-											string(securityinsight.EntitiesMatchingMethodNone),
+											string(securityinsight.MatchingMethodAnyAlert),
+											string(securityinsight.MatchingMethodSelected),
+											string(securityinsight.MatchingMethodAllEntities),
```

Comment:
Just confirmed with the service team and verified in my local: the original enums `All`, `Custom`, `None` corresponds to the `AllEntities`, `Selected` and `AnyAlert`. If the user creats the alert rule using the old API version, and do a `GET` using the new one, the enums are returned in their new form.

Reply:
Reckon for any enums that dont do this could translate them and provide a warning to the user for a few versions - although if all old enums behave like this users will know to fix their configuration when their is idempotency

Reply:
Good idea, I'll make the change, with one things worth mentioning: the warning is only visible in the debug log.